test/unit/puppetlabs/services/jruby/jruby_puppet_service_test.clj

```diff
@@ -38,27 +40,26 @@
 
 (deftest test-error-during-init
   (testing
-      (str "If there as an exception while putting a JRubyPuppet instance in "
-           "the pool the application should shut down.")
+   (str "If there as an exception while putting a JRubyPuppet instance in "
+        "the pool the application should shut down.")
     (logging/with-test-logging
-      (with-redefs [jruby-internal/create-pool-instance!
-                    (fn [& _] (throw (Exception. "42")))]
-                   (let [got-expected-exception (atom false)]
-                     (try
-                       (bootstrap/with-app-with-config
-                         app
-                         default-services
-                         (jruby-service-test-config 1)
-                         (tk/run-app app))
-                       (catch Exception e
-                         (let [cause (stacktrace/root-cause e)]
-                           (is (= (.getMessage cause) "42"))
-                           (reset! got-expected-exception true))))
-                     (is (true? @got-expected-exception)
-                                "Did not get expected exception."))
-                   (is (logged?
-                         #"^shutdown-on-error triggered because of exception!"
-                                :error))))))
+     (with-redefs [jruby-internal/create-pool-instance!
+                   (fn [& _] (throw (Exception. "42")))]
+       (let [got-expected-exception (atom false)]
+         (try
+           ; We use this instead of with-app-with-config because with-app-with-config has an implicit stop
```

Comment:
> Once the HUP changes were in we started seeing a double-stop, which broke the logging check.

Ah, okay.  Took me a while to wrap my head around what this meant.  I think what happens with this test now is that the ShutdownService's `shutdown-on-error` is triggered twice - once for the failure in `prime-pool!` when the initial JRuby instance cannot be created and once again from `swap-and-drain-pool!` via `flush-pool-for-shutdown!` when the service is being stopped.  So it's probably not that the service's `stop` is being called twice that breaks the logging check but that the `shutdown-on-error triggered because of exception` being logged from both of those failed operations now - whereas the service `stop` prior to changes in this PR didn't try to drain the pool.  Since the `logged?` macro only allows you to match one instance - and not two - makes sense that you couldn't use it here anymore.  I'm good with removing the check for the log message.  I generally don't like to have those kinds of assertions in unit tests anyway.

I think this could be reverted back to using the `with-app-with-config` macro now, as you mentioned.  Seems like the remaining assertion for "@got-expected-exception" would work fine here.  Could also just have the try/catch wrapped around the `run-app` call since that's the one we expect to propagate the exception up, right?